File: Number of Laser Beams in a Bank.py
# https://leetcode.com/problems/number-of-laser-beams-in-a-bank/description/?envType=daily-question&envId=2024-01-03
# Number of Laser Beams in a Bank

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        laser = 0
        last_floor = 0

        for idx, floor in enumerate(bank):
            
            if "1" not in floor:
                continue
            else:
                num = floor.count("1")

                if last_floor == 0:
                    last_floor = num
                    continue

                laser += last_floor * num
                last_floor = num

        return laser

File: Number of Laser Beams in a Bank.js
// https://leetcode.com/problems/number-of-laser-beams-in-a-bank/?envType=daily-question&envId=2025-10-27
// Number of Laser Beams in a Bank

/**
 * @param {string[]} bank
 * @return {number}
 */
var numberOfBeams = function(bank) {
    
    let prevDevices = 0;
    let totalBeams = 0;

    for (const row of bank) {
        const devices = [...row].filter(c => c === '1').length;
        if (devices > 0) {
        totalBeams += prevDevices * devices;
        prevDevices = devices;
        }
    }

    return totalBeams;

};